pilot/pkg/config/clusterregistry/conversion.go

```diff
@@ -101,62 +100,140 @@ func (cs *ClusterStore) GetPilotClusters() (clusters []*k8s_cr.Cluster) {
 	return
 }
 
-// ReadClusters reads multiple clusters from files in a directory
-func ReadClusters(crPath string) (cs *ClusterStore, err error) {
+// ReadClusters reads multiple clusters from a OcnfigMap
+func ReadClusters(k8s kubernetes.Interface, configMapName string, configMapNamespace string, registriesDir string) (cs *ClusterStore, err error) {
 	cs = &ClusterStore{
 		clusters: []*k8s_cr.Cluster{},
 		cfgStore: nil,
 	}
-
 	var clusters []*k8s_cr.Cluster
-	err = filepath.Walk(crPath, func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			return err
+
+	// getClustersConfigs
+	clusters, err = getClustersConfigs(k8s, configMapName, configMapNamespace, registriesDir)
+	if err != nil {
+		return nil, err
+	}
+	if len(clusters) == 0 {
+		return nil, fmt.Errorf("no kubeconf found in provided ConfigMap: %s/%s", configMapNamespace, configMapName)
+	}
+
+	// For the time being, assume that ClusterPilotCfgStore is only set for one cluster only.
+	// This cluster will be used as the pilot's config store.
+	for _, cluster := range clusters {
+		log.Infof("ClusterPilotCfgStore: %s", cluster.ObjectMeta.Annotations[ClusterPilotCfgStore])
+		if isCfgStore, _ := strconv.ParseBool(cluster.ObjectMeta.Annotations[ClusterPilotCfgStore]); isCfgStore {
+			if cs.cfgStore != nil {
+				err = fmt.Errorf("multiple cluster config stores are defined")
+				log.Warnf("%v", err)
+				return nil, err
+			}
+			cs.cfgStore = cluster
+		} else {
+			cs.clusters = append(cs.clusters, cluster)
 		}
-		if !supportedExtensions[filepath.Ext(path)] || (info.Mode()&os.ModeType) != 0 {
-			return nil
+	}
+	if cs.cfgStore == nil {
+		log.Warnf("no config store is defined in the cluster registries")
+		return nil, nil
+	}
+
+	return
+}
+
+// getClustersConfigs(configMapName,configMapNamespace)
+func getClustersConfigs(k8s kubernetes.Interface, configMapName, configMapNamespace string, registryDir string) ([]*k8s_cr.Cluster, error) {
+	clusters := []*k8s_cr.Cluster{}
+	clusterRegistry, err := k8s.CoreV1().ConfigMaps(configMapNamespace).Get(configMapName, metav1.GetOptions{})
+	if err != nil {
+		return []*k8s_cr.Cluster{}, err
+	}
+	obj := k8s_cr.Cluster{}
+	for key, data := range clusterRegistry.Data {
+		fmt.Printf("key: %s data: %+v\n", key, data)
```

Comment:
log...

Reply:
Removed, it was a leftover from debugging.